pkg/client/events.go

```diff
@@ -73,6 +74,29 @@ func (e *events) Create(event *api.Event) (*api.Event, error) {
 	return result, err
 }
 
+// Update modifies an existing event. It returns the copy of the event that the server returns,
+// or an error. The namespace and key to update the event within is deduced from the event. The
+// namespace must either match this event client's namespace, or this event client must have been
+// created with the "" namespace. Update also requires the ResourceVersion to be set in the event
+// object.
+func (e *events) Update(event *api.Event) (*api.Event, error) {
+	if e.namespace != "" && event.Namespace != e.namespace {
```

Comment:
This check should be done on the server - there's really no need to do it in the client (it indicates a coding error in most cases).